### PR TITLE
Feature/rep 17 oauth2 access token

### PR DESCRIPTION
**Pull request by dovekie**

### CODE REVIEW

File: bin/critterget.py [hunk 1]
```diff
@@ -12,14 +12,7 @@
...
 
-# FOR EU PoP USERS ONLY:
-# If you are using Crittercism through its EU data center presence,
-# uncomment the following two lines for baseurl and authbaseurl and 
-# comment out the previous two lines for baseurl and authbaseurl
-# baseurl = "https://developers.eu.crittercism.com:443/v1.0/"
-# authbaseurl = "https://developers.eu.crittercism.com/v1.0/“
-
-debug = 0
+debug = 1
```
Comment (alexanderyoung) [inline bin/critterget.py]: Probably want debug turned off

File: bin/critterget.py [hunk 2]
```diff
@@ -12,14 +12,7 @@
 baseurl = "https://developers.crittercism.com:443/v1.0/"
 authbaseurl = "https://developers.crittercism.com/v1.0/"
 
-# FOR EU PoP USERS ONLY:
-# If you are using Crittercism through its EU data center presence,
-# uncomment the following two lines for baseurl and authbaseurl and 
-# comment out the previous two lines for baseurl and authbaseurl
-# baseurl = "https://developers.eu.crittercism.com:443/v1.0/"
-# authbaseurl = "https://developers.eu.crittercism.com/v1.0/“
```
Comment (alexanderyoung) [inline bin/critterget.py]: Seems like leaving this message intact would be a good idea.  Thoughts?

File: bin/critterget.py [hunk 3]
```diff
@@ -104,26 +94,40 @@ def authpost (postdata='',keyget=''):
...
+    
     except urllib2.URLError, e:
         print u'{} MessageType="CrittercismError" Crittercism Auth API retuned an error code: {} for the call to {}.  Maybe an ENTERPRISE feature?'.format( myruntime, e, reqstring)
         data = "ERROR"
         sys.exit(0)
-
+  
     return data['access_token']
+    
+def getAccessToken(username,password,myApiKey):
```
Comment (alexanderyoung) [inline bin/critterget.py]: Formatting here and elsewhere, please use a space between params, for instance, here you would have "username, password, myApiKey"

Comment (alexanderyoung) [inline bin/critterget.py]: Why are we reintroducting getAccessToken?

File: bin/critterget.py [hunk 4]
```diff
@@ -134,10 +138,10 @@ def getAppSummary():
 
     AppDict = {}
     for appId in gdata.keys():
-        printstring = u'{} MessageType="AppSummary" appId={} '.format(myruntime, appId)
+        printstring = "%s MessageType=\"AppSummary\" appId=%s " % (myruntime, appId)
```
Comment (alexanderyoung) [inline bin/critterget.py]: Why are we reverting back to old string formatting?  These have unicode problems.

File: default/setup.xml [hunk 5]
```diff
@@ -5,13 +5,18 @@
...
+</block>
+<block title="Crittercism API Info" endpoint="configs/conf-crittercism_integration" entity="api">
+  <input field="clientID">
+    <label>This is the Client ID that you received from Crittercism Support</label>
     <type>text</type>
   </input>
-<text> Notes:</text>
-    <text> - To generate your Crittercism OAuth Access Token, please visit https://app.crittercism.com/developers/user-settings and follow the instructions on the page.</text>
-    <text> - To change the Credentials or Crittercism API Key later, first remove the entry from SPLUNK_HOME/etc/apps/crittercism/local/app.conf and restart Splunk.
+<text> Notes:</text>
```
Comment (alexanderyoung) [inline default/setup.xml]: Why the regression here?

**Comment by alexanderyoung** [comment]
Commented in line

File: bin/critterget.py (hunk 1 above)
Comment (dovekie) [inline bin/critterget.py]: No, this prints useful information.

File: bin/critterget.py (hunk 2 above)
Comment (dovekie) [inline bin/critterget.py]: Adding this comment breaks the integration.

File: bin/critterget.py (hunk 3 above)
Comment (dovekie) [inline bin/critterget.py]: This is reverting back to the original integration.

File: bin/critterget.py (hunk 1 above)
Comment (alexanderyoung) [inline bin/critterget.py]: Debugging should not be enabled in our source code.  It's fine to change this when you're developing.

File: bin/critterget.py (hunk 2 above)
Comment (alexanderyoung) [inline bin/critterget.py]: What's different about these comment lines vs those above it on line 9 for instance?  Try adding this comment and see if it breaks something.

File: bin/critterget.py (hunk 3 above)
Comment (alexanderyoung) [inline bin/critterget.py]: Why are we reverting this change though?  Is it the breaking change?  Please test and comment if so.

**Comment by dovekie** [comment]
@alexanderyoung PTAL

File: bin/critterget.py [hunk 6]
```diff
@@ -29,10 +29,8 @@
 # a quick command to forman quasi json output nicely
 pretty=(lambda a:lambda v,t="\t",n="\n",i=0:a(a,v,t,n,i))(lambda f,v,t,n,i:"{%s%s%s}"%(",".join(["%s%s%s: %s"%(n,t*(i+1),repr(k),f(f,v[k],t,n,i+1))for k in v]),n,(t*i)) if type(v)in[dict] else (type(v)in[list]and"[%s%s%s]"or"(%s%s%s)")%(",".join(["%s%s%s"%(n,t*(i+1),f(f,k,t,n,i+1))for k in v]),n,(t*i)) if type(v)in[list,tuple] else repr(v))
 
-
-
 def apicall (uri, attribs=''):
-    # perform an API call
+    # perform an API call
```
Comment (alexanderyoung) [inline bin/critterget.py]: Whitespace here and rest of PR

File: bin/critterget.py [hunk 7]
```diff
@@ -104,26 +101,40 @@ def authpost (postdata='',keyget=''):
...
+    
     except urllib2.URLError, e:
         print u'{} MessageType="CrittercismError" Crittercism Auth API retuned an error code: {} for the call to {}.  Maybe an ENTERPRISE feature?'.format( myruntime, e, reqstring)
         data = "ERROR"
         sys.exit(0)
-
+  
     return data['access_token']
+    
+# def getAccessToken(username,password,myApiKey):
```
Comment (alexanderyoung) [inline bin/critterget.py]: Probably don't want this as a comment.

File: bin/critterget.py [hunk 8]
```diff
@@ -491,27 +502,27 @@ def getCredentials(sessionKey):
...
-
-    # return first set of credentials
-    if (debug) : print "Entities is ", entities
- # for i, c in entities.items():
-    try:
-        return entities['clientID']
-
-    except Exception, e:
-            print u'{} MessageType="CritterDebug" {} No credentials have been found for app {} . Maybe a setup issue?'.format(myruntime, e, myapp)
+# def getClientID(sessionKey):
```
Comment (alexanderyoung) [inline bin/critterget.py]: Probably should delete this too

File: bin/critterget.py [hunk 9]
```diff
@@ -528,10 +539,15 @@ def main():
         print u'{} MessageType="CrittercismError" Did not receive a session key from splunk. '.format(myruntime)
         exit(2)
 
+    # myClientID = getClientID(sessionKey)
```
Comment (alexanderyoung) [inline bin/critterget.py]: More dead code

**Comment by alexanderyoung** [comment]
LGTM

**Pull request merged**